packages/components/Button/src/useButton.ts

```diff
@@ -56,14 +56,22 @@ export const useButton = (props: ButtonProps): ButtonInfo => {
   );
   const onKeyPress = React.useCallback(
     (e) => {
-      if (isProcessingKeyboardInvocation) {
-        onClick?.(e);
-        isProcessingKeyboardInvocation = false;
+      if (shouldOnlyFireIfPressed) {
+        if (isProcessingKeyboardInvocation) {
+          onClick?.(e);
+          isProcessingKeyboardInvocation = false;
+        }
+      } else {
+        if (Platform.OS === 'macos') {
+          // Do nothing as macOS's pressable already calls onPress with onKeyDown
```

Comment:
Come to think of it, shouldn't we be calling `onKeyDown` inside here, not `onClick`? AKA, the user-provided prop for what to do on key down?

Reply:
Ah, I see. I still don't get why win32 doesn't have the same problem of double calling `onClick` (we tested it), and why the _isKeyDown prevents it. But 🤷🏾. I'm gonna merge this now because of the time pressure.

Reply:
So Win32 dispatches the click in key up (unlike Mac which dispatches the click in key down). IIRC from my own investigation, `_isKeyDown` isn't set in the key up handling on Win32.

Reply:
That.... sounds like a bug? But one that inadvertently hides this other bug of double click handling?

Reply:
checking the state of `_keyDown` in the key up callback? Yeah, that did seem suspect to me, but I didn't spend time trying to figure out what was the intention behind it.

Reply:
The intention is to make sure _isKeyDown_ was set, so that keyDown was invoked before keyUp. There's an edge case where you can keyDown button1, Tab to move focus, and keyUp button2. In that case, you don't want button2 to fire onPress, but it will. The fix was to check that keyDown was called before keyUp via the boolean.